src/jit/inline.h

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _INLINE_H_
+#define _INLINE_H_
+
+// InlineDecision describes the various states the jit goes through when
+// evaluating an inline candidate. It is distinct from CorInfoInline
+// because it must capture interal states that don't get reported back
+// to the runtime.
+
+enum class InlineDecision {
+    UNDECIDED,
+    CANDIDATE,
+    SUCCESS,
+    FAILURE,
+    NEVER
+};
+
+// Possible targets of an inline observation
+
+enum class InlineTarget
+{
+    CALLEE,         // observation applies to all calls to this callee
+    CALLER,         // observation applies to all calls made by this caller
+    CALLSITE        // observation applies to a specific call site
+};
+
+// Possible impact of an inline observation
+
+enum class InlineImpact
+{
+    FATAL,          // inlining impossible, unsafe to evaluate further
+    SERIOUS,        // inlining impossible for fundamental reasons, deeper explanation safe
```

Comment:
I'm not sure I could come up with better terms, but "SERIOUS" seems pretty vague to me. How about "FUNDAMENTAL" - you used it in the definition, and it seems more meaningful to me (though longer)?